### PR TITLE
Fix multimap issues.

### LINKED ISSUE #17: Creating and destroying multiple maps causing other maps to throw PlatformException(channel-error, Unable to establish connection on channel., null, null)
When we have initial screen which contains map, and we push and pop a route with other map, calling functions on map which is on initial screen throws PlatformException. Generally creating and destroying map even without pushing and popping routes is causing the same exception. For example, we have a widget which is initially off screen, and when user clicks a button, widget slides in and displays a map which is destroyed when the user is finished with that widget.

Reproduction code:

```
void main() {
  runApp(const MaterialApp(
    home: MyApp(),
  ));
}

class MyApp extends StatefulWidget {
  const MyApp({super.key});

  @override
  State<MyApp> createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  MapboxMap? _map;
  bool _styleLoaded = false;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('First Map'),
      ),
      body: Stack(
        children: [
          MapWidget(
            resourceOptions: ResourceOptions(accessToken: <add access token>),
            styleUri: MapboxStyles.LIGHT,
            onMapCreated: (MapboxMap map) => _map = map,
            onStyleLoadedListener: (StyleLoadedEventData event) => _styleLoaded = true,
          ),
          Positioned(
            left: 16,
            bottom: 16,
            child: ElevatedButton(
              onPressed: () => Navigator.push(
                context,
                MaterialPageRoute(builder: (context) => const SecondRoute()),
              ),
              child: const Text('Second map'),
            ),
          ),
          Positioned(
            right: 16,
            bottom: 16,
            child: ElevatedButton(
              onPressed: () {
                if(_styleLoaded) {
                  _map!.style.styleLayerExists('layer_id');
                }
              },
              child: const Text('Map action'),
            ),
          ),
        ],
      ),
    );
  }
}

class SecondRoute extends StatelessWidget {
  const SecondRoute({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Second Map'),
      ),
      body: MapWidget(
        resourceOptions: ResourceOptions(accessToken: <add access token>),
        styleUri: MapboxStyles.LIGHT,
      ),
    );
  }
}
```

Steps to reproduce:
1. Click on Second Map button.
2. Go back to previous screen.
3. Click on Map action button.

Following exception is thrown:

```
E/flutter (27096): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: PlatformException(channel-error, Unable to establish connection on channel., null, null)
E/flutter (27096): #0      StyleManager.styleLayerExists (package:mapbox_maps_flutter/src/pigeons/map_interfaces.dart:5766:7)
E/flutter (27096): <asynchronous suspension>
```

### PR DESCRIPTION
**Pull request by yunikkk**
Pigeon does not support generating separate instances of api [at the moment](https://github.com/flutter/flutter/issues/66710). 
Thus to separate the channels I've appended suffix ("/map_${id}") at the BinaryMessenger level in ProxyBinaryMessenger.

Fixes #17, #54

### CODE REVIEW

**Comment by yunikkk** [comment]
@LjubiTech-Maxko, @dawid-niedzwiecki could you also check if the fix helps on your end?

**Comment by dawid-niedzwiecki** [comment]
> @LjubiTech-Maxko, @dawid-niedzwiecki could you also check if the fix helps on your end?

I'll check if that helps, thank you very much! I'll post an update in about 2 hours.

**Comment by dawid-niedzwiecki** [comment]
> @LjubiTech-Maxko, @dawid-niedzwiecki could you also check if the fix helps on your end?

I imported this branch:

```
mapbox_maps_flutter:
  git:
    url: https://github.com/mapbox/mapbox-maps-flutter.git
    ref: yds-fix-multimap-issues
```

And ran a `flutter pub get`. When running in debug, I get the following error:  

```
Launching lib/main.dart on moto g 10 in debug mode...
main.dart:1
e: /Users/niedzwiecki/.pub-cache/git/mapbox-maps-flutter-163b948a2ed20e4d64c7ab6ae4aba4cb5344faa8/android/src/main/kotlin/com/mapbox/maps/mapbox_maps/MapboxMapController.kt: (17, 7): Redeclaration: MapboxMapController
e: /Users/niedzwiecki/.pub-cache/git/mapbox-maps-flutter-163b948a2ed20e4d64c7ab6ae4aba4cb5344faa8/android/src/main/kotlin/com/mapbox/maps/mapbox_maps/MapboxMapFactory.kt: (12, 7): Redeclaration: MapboxMapFactory
e: /Users/niedzwiecki/.pub-cache/hosted/pub.dev/mapbox_maps_flutter-0.4.0/android/src/main/kotlin/com/mapbox/maps/mapbox_maps/MapboxMapController.kt: (17, 7): Redeclaration: MapboxMapController
e: /Users/niedzwiecki/.pub-cache/hosted/pub.dev/mapbox_maps_flutter-0.4.0/android/src/main/kotlin/com/mapbox/maps/mapbox_maps/MapboxMapFactory.kt: (12, 7): Redeclaration: MapboxMapFactory

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':mapbox_maps_flutter:compileDebugKotlin'.
> Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 47s
Exception: Gradle task assembleDebug failed with exit code 1
Exited
```

**Comment by LjubiTech-Maxko** [comment]
@dawid-niedzwiecki Try doing `flutter clean` before running application.

@yunikkk I tested this on sample code posted in the mentioned issue and our own app and can confirm this resolves the issues. Although, I have to say that I haven't tested all the features that should be affected with this, for example annotation managers.

**Comment by dawid-niedzwiecki** [comment]
I did a `flutter pub cache clean` and a `flutter pub get` and it built.

Regarding the PR - works perfectly fine in my case as well! I am very grateful! Thanks!

File: android/src/main/kotlin/com/mapbox/maps/mapbox_maps/ProxyBinaryMessenger.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import java.nio.ByteBuffer
+
+// BinaryMessenger that proxies all calls to another BinaryMessenger,
+// but with a suffix appended to the channel name.
+// Different map instances use different suffixes to avoid channel name conflicts.
+class ProxyBinaryMessenger(
+  private val messenger: BinaryMessenger,
+  private val nameSuffix: String,
+) : BinaryMessenger {
+
```
Comment (alexander-kulikovskii) [inline android/src/main/kotlin/com/mapbox/maps/mapbox_maps/ProxyBinaryMessenger.kt:13]: nit:
private val channelName = "$channel$nameSuffix"

File: android/src/main/kotlin/com/mapbox/maps/mapbox_maps/ProxyBinaryMessenger.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  override fun send(channel: String, message: ByteBuffer?) {
+    messenger.send("$channel$nameSuffix", message)
```
Comment (alexander-kulikovskii) [inline android/src/main/kotlin/com/mapbox/maps/mapbox_maps/ProxyBinaryMessenger.kt:15]:
```suggestion
    messenger.send(channelName, message)
```

Comment (yunikkk) [reply android/src/main/kotlin/com/mapbox/maps/mapbox_maps/ProxyBinaryMessenger.kt:15]: Cant extract it here - channel is a function call parameter and changes.

Comment (alexander-kulikovskii) [reply android/src/main/kotlin/com/mapbox/maps/mapbox_maps/ProxyBinaryMessenger.kt:15]: ah, yes
you can add some like that:
```
private fun String.toChannelName() = "$this$nameSuffix"
```
inside class. just not to forgot to change each place when you will make some refactoring in the future

**Review by alexander-kulikovskii — APPROVED** [review]

File: android/src/main/kotlin/com/mapbox/maps/mapbox_maps/ProxyBinaryMessenger.kt (hunk 2 above)
Comment (yunikkk) [reply android/src/main/kotlin/com/mapbox/maps/mapbox_maps/ProxyBinaryMessenger.kt]: Done

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,7 +1,15 @@
...
 
 ### Common
 * Expose `package:turf/helpers.dart`.
+=======
+## 0.4.1 
+
+### Common
+* Bump platform Maps SDK dependencies to 10.10.0.
+* Fix issue with multiple maps overriding platform channels of the previous instances.
+>>>>>>> c718dda (Update changelog, bump platform Maps SDK version.)
```
Comment (alexander-kulikovskii) [inline CHANGELOG.md:12]: rm?

Comment (yunikkk) [reply CHANGELOG.md]: Yep, hadn't added rebase fix to the commit

**Pull request merged**